boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/IAMStarter.java

```diff
@@ -0,0 +1,156 @@
+            throws InterruptedException, IOException, ExecutionException {
+
+        if (!started && buildJavaCommand(paHome, jvmArgs) && buildMicroservicePath(paHome, microservicesPath)) {
+
+            LOGGER.info("Starting IAM microservice");
+
+            ProcessBuilder processBuilder = new ProcessBuilder(command);
+            processBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
+
+            process = processBuilder.start();
+            LOGGER.info(streamOutput(process.getInputStream()));
+
+            started = true;
+        }
+        return process;
+    }
+
+    /**
+     * Stream microservice output (using an executor) to check that it starts properly
+     */
+    private static String streamOutput(InputStream inputStream)
+            throws IOException, InterruptedException, ExecutionException {
+
+        // Stream microservice output
+        BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        Future<String> future = executor.submit(new Callable() {
+
+            public String call() throws IOException {
+
+                try {
+                    String line = null;
+
+                    while ((line = br.readLine()) != null) {
+                        System.out.print(".");
+                        if (line.contains(READY_MARKER)) {
+                            break;
+                        }
+                    }
+                } finally {
+                    br.close();
+                }
+                return "\nIAM Microservice started";
+            }
+        });
+
+        // Check for timeout
+        try {
+            return future.get(TIMEOUT, TimeUnit.SECONDS);
+        } catch (TimeoutException toe) {
+            // Stop streaming the output, but the microservice continues to execute
+            br.close();
+            LOGGER.warn("Warning: IAM Microservice timed out to start. See the logs for the details.");
+            return "\nWarning: IAM Microservice timed out to start. See the logs for the details.";
+        } finally {
+            executor.shutdownNow();
+            future.cancel(true);
+        }
+    }
+
+    /**
+     * Prepare java command with jvm args
+     */
+    private static boolean buildJavaCommand(String pa_home, String[] jvmArgs) {
+
+        String javaCmd = pa_home + SEPARATOR + "jre" + SEPARATOR + "bin" + SEPARATOR + "java";
+
+        if (new File(javaCmd).exists()) {
+            command.add(javaCmd);
+            command.add("-jar");
+            command.addAll(Arrays.asList(jvmArgs));
+            return true;
+        } else {
+            System.out.println("Java command not found when starting IAM microservice");
+            return false;
+        }
+
+    }
+
+    /**
+     * Check the microservice executable war
+     */
+    private static boolean buildMicroservicePath(String pa_home, String microservices_path) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)